### PR TITLE
Adding IP address to BQ table

### LINKED ISSUE #9: Add user's IP address to BQ
- IP address sent back frontend.
- Needs to be added as new column in BQ.

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**
- SQS message has user's IP address which gets added to the BQ table. 
https://console.cloud.google.com/bigquery?authuser=0&project=avantifellows&ws=!1m10!1m4!4m3!1savantifellows!2sauth_logs!3sstaging-attendance-logs!1m4!1m3!1savantifellows!2sbquxjob_3dbcb634_183c099d63f!3sasia-south1

- Frontend PR - https://github.com/avantifellows/Portal/pull/70

### CODE REVIEW

File: lambda_function.py [hunk 1]
```diff
@@ -21,42 +21,49 @@ def lambda_handler(event, lambda_context):
...
-            row = {}
-
             # check if the key values exist in the message sent
             # even if one field is missing,
             #   the row isn't inserted and an error is logged
-
             if all(
-                (k in message for k in ("dateTime", "purpose", "authType", "user"))
+                (k in message for k in ("dateTime", "purpose", "authType",
+                 "user", "program", "userType", "sessionId", "userData"))
```
Comment (pritamps) [inline lambda_function.py:39]: This seems to be adding more than just the IP? Why all these new variables?

File: lambda_function.py [hunk 2]
```diff
@@ -21,42 +21,49 @@ def lambda_handler(event, lambda_context):
...
-                    row["user_data_validated"] = each["valid"]
-
-                    return insert_data(row)
+                    row["user_id"] = user_values[i]["userID"]
+                    row["user_data_validated"] = user_values[i]["valid"]
+                    row["number_multiple_entries"] = user_values_length
+                    row["program"] = message["group"]
+                    row["userType"] = message["userType"]
+                    row["session_id"] = message["sessionId"]
+                    row["user_ip_address"] = message["userNetworkData"]["userIp"]
```
Comment (pritamps) [inline lambda_function.py:64]: Ah, maybe just pass only the IP address here since that's all that's needed?

**Review by pritamps — CHANGES_REQUESTED** [review]

File: lambda_function.py (hunk 2 above)
Comment (deepansh96) [reply lambda_function.py:64]: yeah even I am a little confused, this PR is then not JUST for the IP address logging right?
All these other variables will also be logged?

Comment (sudheshna-donthineni) [reply lambda_function.py:64]: So there was no CD process in place before. I was deploying it manually from my computer. So this repo was a little outdated. I have created a PR for the CD process to keep the code up to date.

Comment (deepansh96) [reply lambda_function.py:64]: Ohh got it

**Review by deepansh96 — APPROVED** [review]

**Pull request merged**